src/sieppari/core.cljc

```diff
@@ -37,15 +40,18 @@
         (assoc ctx :stack (iter stack))
         (recur (-> ctx
                    (assoc :queue (pop queue))
-                   (assoc :stack (conj stack interceptor))
+                   (assoc :stack #?(:clj (conj stack interceptor)
+                                    :cljs (doto (or stack (array))
+                                            (.unshift interceptor))))
```

Comment:
Why use JS arrays here instead of linked lists, like the Clojure side does?

Reply:
I am basically using an iterator like the implementation, it can be backed by anything really. I picked native arrays because probably faster in JavaScript but haven't done any performance testing.

Reply:
I'm a bit skeptical of this optimization, but let's have it. `:stack` is kind of part of the interface of Sieppari, since it's exposed in the context map, but on the other hand it's an iterator when leaving, so it's not like you're supposed to inspect it.

Reply:
I had exactly the same thought!